azure-proto-core-test/SystemAssignedIdentityTests.cs

```diff
@@ -0,0 +1,299 @@
+﻿using azure_proto_core.Resources;
+using NUnit.Framework;
+using System;
+using System.IO;
+using System.Linq;
+using System.Text.Json;
+
+namespace azure_proto_core_test
+{
+    public class SystemAssignedIdentityTests : SystemAssignedIdentity
+    {
+        [TestCase(0, null, null, null, null)]
+        [TestCase(0, "72f988bf-86f1-41af-91ab-2d7cd011db47", "de29bab1-49e1-4705-819b-4dfddceaaa98", "72f988bf-86f1-41af-91ab-2d7cd011db47", "de29bab1-49e1-4705-819b-4dfddceaaa98")]
+        [TestCase(1, "72f988bf-86f1-41af-91ab-2d7cd011db47", "de29bab1-49e1-4705-819b-4dfddceaaa98", null, null)]
+        [TestCase(1, "72f988bf-86f1-41af-91ab-2d7cd011db47", "de29bab1-49e1-4705-819b-4dfddceaaa98", "72f988bf-86f1-41af-91ab-2d7cd011db46", "de29bab1-49e1-4705-819b-4dfddceaaa97")]
```

Comment:
I think it would be useful to have test cases in which both PrincipalId and TenantId are non-null, and:
- Tid1 == Tid2, Pid1 < P id2 (result -1)
- Tid1 == Tid2, Pid1 > Pid2 (result 1)
- Tid1 < Tid2, Pid1 == Pid2 (result -1)
- Tid1 > Tid1, Pid1 == Pid2 (result 1)
- Tid1 < Tid2, Pid1 > Pid2 (result -1)
- Tid1 > Tid2, Pid1 < Pid2 (result 1)

Essentially, think of completing a table like:


| CompareTo(Tid1, Tid2) | CompareTo(Pid1, Pid2 | Result |
| ---: | ---------------------------------: | -----: |
| -1 | -1 | -1 |
| -1 |  0 | -1 |
| -1 | 1 | -1|
| 0 | -1 | -1 |
| 0 | 0 | 0 |
| 0 | 1 | 1 |
| 1 | -1 | 1 |
| 1 | 0 | 1 |
|1 | 1 | 1 |